### PR TITLE
Fix misleading exception message, cleanup code

### LINKED ISSUE #24817: FileSystemWatcher misleading exception message
I encountered an exception being thrown in an ASP.NET Core 2.0 project which said "The directory name {0} is invalid." and it was misleading. [`FileSystemWatcher`](https://github.com/dotnet/corefx/blob/724ddb8fc95a5b1f639cfa508a32b9a8c50c1949/src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs#L108) throws an exception with this message, and in my case the path was valid but the directory did not exist.

"The directory name {0} does not exist." would be a more helpful message for me.

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
I'm newbie(on contributing), i grab the issue and try to help. Fixes #24817

### CODE REVIEW

**Comment by JonHanna** [comment]
Tip: If this completely fixes the issue (or at least you believe it does) you can include `Fixes #24817` or `Closes #24817` in the description of the PR and merging it will automatically close the issue. If there would still be some more work to do then including something like `contributes to #24817` would both link to the issue and add a link to here from that issue, which would make finding that some work had been done on it more discoverable by someone looking at the issue.

**Comment by MarcoRossignoli** [comment]
Thank you very much @JonHanna!

**Comment by JonHanna** [comment]
NP. See https://help.github.com/articles/closing-issues-using-keywords/ for more variants.

**Comment by MarcoRossignoli** [comment]
@dotnet-bot test  OSX x64 Debug Build

File: src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs [hunk 1]
```diff
@@ -105,8 +105,8 @@ public FileSystemWatcher(string path, string filter)
                 throw new ArgumentNullException(nameof(filter));
 
             // Early check for directory parameter so that an exception can be thrown as early as possible.
-            if (path.Length == 0 || !Directory.Exists(path))
-                throw new ArgumentException(SR.Format(SR.InvalidDirName, path), nameof(path));
+            if (!Directory.Exists(path))
```
Comment (ViktorHofer) [inline src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs]: IMO the check for `path.Length == 0` should stay and output the current error message and after it the directory exists check with the more concrete error message.

File: src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Directory.Create.cs [hunk 2]
```diff
@@ -9,6 +9,9 @@ namespace System.IO.Tests
 {
     public class Directory_Create_Tests : FileSystemWatcherTest
     {
+
```
Comment (ViktorHofer) [inline src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Directory.Create.cs]: nit: empty lines.

File: src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Directory.Create.cs [hunk 3]
```diff
@@ -80,5 +83,31 @@ public void FileSystemWatcher_Directory_Create_SymLink()
                 ExpectEvent(watcher, WatcherChangeTypes.Created, action, cleanup, symLinkPath);
             }
         }
+
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "On Desktop the message is different")]        
+        public void FileSystemWatcher_Directory_NotExists()
```
Comment (ViktorHofer) [inline src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.Directory.Create.cs]: this test won't work when we start to localize .NET Core. We normally don't assert full error messages. I recommend removing this test and check if the current test scenario tests for both empty string and a not existing path.

File: src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.File.Create.cs [hunk 4]
```diff
@@ -9,7 +9,7 @@
 namespace System.IO.Tests
 {
     public class File_Create_Tests : FileSystemWatcherTest
-    {
+    {
```
Comment (ViktorHofer) [inline src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.File.Create.cs]: please revert changes here.

File: src/System.IO.FileSystem.Watcher/tests/System.IO.FileSystem.Watcher.Tests.csproj [hunk 5]
```diff
@@ -8,7 +8,9 @@
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netstandard-Linux-Release|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netstandard-OSX-Debug|AnyCPU'" />
   <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netstandard-OSX-Release|AnyCPU'" />
-  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netstandard-Windows_NT-Debug|AnyCPU'" />
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netstandard-Windows_NT-Debug|AnyCPU'">
```
Comment (ViktorHofer) [inline src/System.IO.FileSystem.Watcher/tests/System.IO.FileSystem.Watcher.Tests.csproj]: please revert all the changes in this file.

**Comment by MarcoRossignoli** [comment]
@ViktorHofer i've added two test for empty path and inexistent path, but there is "FileSystemWatcher_ctor_InvalidStrings" in "FileSystemWatcher.unit.cs" with similar tests, maybe my tests are redundant.

File: src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.File.Create.cs (hunk 4 above)
Comment (MarcoRossignoli) [reply src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.File.Create.cs]: I need help here, i don't see any change to revert on this file on my source(VS), can you tell me how to revert this(maybe on this specific case so i can learn how to do for future)?

**Comment by MarcoRossignoli** [comment]
@dotnet-bot test Windows x86 Release Build

File: src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.File.Create.cs (hunk 4 above)
Comment (danmosemsft) [reply src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.File.Create.cs]: It's a whitespace only change. You need to configure your diff tool to show whitespace only changes. Github also showed them by default.

Comment (danmosemsft) [reply src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.File.Create.cs]: In this case you can just revert this file. Then all looks good, thanks!

Comment (ViktorHofer) [reply src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.File.Create.cs]: sry, I didn't notice your response: `git checkout -- path/to/file`

**Comment by MarcoRossignoli** [comment]
@ViktorHofer the command doesn't work for me(do nothing), so i replaced the file from master. @danmosemsft i found the "white space" settings on VS merge tools thank you!

**Comment by danmosemsft** [comment]
Thanks @MarcoRossignoli . Perhaps there are other error messages elsewhere in CoreFX that could be made more precise, or include missing useful informatoin?

**Comment by MarcoRossignoli** [comment]
@danmosemsft  it's a pleasure to help. I'll try to search other cases and after grab other issues. Thanks to all of you for helping!

**Pull request merged**